### ISSUE TITLE
luajit v2.1-20170405 (openresty 1.11.2.3) - unexpected symbol near 'pragma'

### ISSUE THREAD

**Issue by alonbg** [openresty/luajit2#10]
Reproduced on openresty_1.11.2.3/luajit/bin/luajit with https://github.com/luapower/blake2/blob/master/blake2.lua

blake2.lua:125: unexpected symbol near 'pragma' at line 14

**Does not** reproduce on openresty/luajit2 tag v2.1-20170405 without the `-msse4.2` compilation flags or [latest commit](https://github.com/openresty/luajit2/commit/9ccc8867b8e520906271ca36e954032054b07e33) excluded.

compiled on Intel(R) Xeon(R) CPU E3-1225 V2 @ 3.20GHz (supports SSE4.2)

**Comment by alonbg** [comment]
minimal reproduction code:

```
local ffi = require'ffi'
ffi.cdef[[

#pragma pack(push, 1)
]]
```

**Comment by agentzh** [comment]
@alonbg Have you tried out the latest `v2.1` branch of the official luajit git repository? I'm pretty sure we never touched that part of luajit in our branch.

**Comment by alonbg** [comment]
Hi @agentzh , yes I tried the  latest v2.1 with the -msse4.2 flag - doesn't reproduce.

**Comment by agentzh** [comment]
@alonbg Thanks for the report! I can reproduce it on my side.

@yangshuxin Will you please take a look at this? It seems that it is triggered by our `lj_str_new` optimizations. Thanks!

**Comment by yangshuxin** [comment]
Turn out the string hash is hard-coded in parser.

{code}
1763 static void cp_decl_multi(CPState *cp)                                          
1764 {                                                                               
...
1779       } else if (tok == CTOK_IDENT &&                                           
1780                  cp->str->hash == H_(187aab88,fcb60b42)) { /* line */           
...
1791     } 
{code}

**Comment by yangshuxin** [comment]
A proposed fix patch is just sent to @agentzh, since I will going to travel this Wed morning. I wonder if @agentzh can take over this bug in case this is urgent.

**Comment by yangshuxin** [comment]
I'm returning back, will make a PR ASAP.

**Comment by agentzh** [comment]
@alonbg Will you try @yangshuxin's LuaJIT patch in #11 on your side? Thanks!

**Comment by alonbg** [comment]
@agentzh - yes. I'll update ..

**Comment by agentzh** [comment]
@alonbg I believe this is already fixed in the latest v2.1-agentzh branch. I'm closing this.
